File: networklogger/src/main/java/com/network/logger/util/EditTextDelay.java
package com.network.logger.util;

import android.text.Editable;
import android.text.TextWatcher;

import androidx.fragment.app.FragmentActivity;

import java.util.Timer;
import java.util.TimerTask;

public class EditTextDelay implements TextWatcher {

    private final FragmentActivity context;
    private Timer timer = new Timer();

    public Listener listener;

    public interface Listener {
        void onTextChanged(String data);
    }

    public TextWatcher setListener(Listener listener) {
        this.listener = listener;
        return this;
    }

    public EditTextDelay(FragmentActivity context) {
        this.context = context;
    }

    @Override
    public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

    }

    @Override
    public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {

    }

    @Override
    public void afterTextChanged(final Editable editable) {
        timer.cancel();
        timer = new Timer();
        timer.schedule(
                new TimerTask() {
                    @Override
                    public void run() {
                        if (context == null) return;
                        context.runOnUiThread(new Runnable() {
                            @Override
                            public void run() {
                                listener.onTextChanged(editable.toString().trim());
                            }
                        });
                    }
                }, 2000);
    }
}

File: networklogger/src/main/java/com/network/logger/util/EditTextDelay.kt
package com.network.logger.util

import android.text.Editable
import android.text.TextWatcher
import androidx.fragment.app.FragmentActivity
import java.util.Timer
import java.util.TimerTask

class EditTextDelay(private val context: FragmentActivity?) : TextWatcher {
    private var timer = Timer()

    var listener: Listener? = null

    interface Listener {
        fun onTextChanged(data: String?)
    }

    fun setListener(listener: Listener?): TextWatcher {
        this.listener = listener
        return this
    }

    override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
    }

    override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
    }

    override fun afterTextChanged(editable: Editable) {
        timer.cancel()
        timer = Timer()
        timer.schedule(
            object : TimerTask() {
                override fun run() {
                    if (context == null) return
                    context.runOnUiThread(Runnable {
                        listener?.onTextChanged(
                            editable.toString().trim { it <= ' ' })
                    })
                }
            }, 1000
        )
    }
}
